### PR TITLE
A4A: disable global notifications

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Related to pfSQfS-Bd-p2

## Proposed Changes

In Automattic for Agencies we currently don't have a **masterbar** and don't wire up WPCOM notification bar for our clients. Thus, to improve overall performance and reduce number of unnecessary calls we want to remove Global Notifications for now.

## Why are these changes being made?

To reduce number of redundant calls of improve page load time.

## Testing Instructions

- Check live links bellow and inspect dev console Network tab
- For A4A, no `notifications` calls should be present
- For Jetpack Cloud and Wordpress.com all should work as before and Wordpress.com masterbar should work properly

Screenshot 2024-12-20 at 9 07 41 AM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by travisw — APPROVED** [review]
Looks good. Nice little optimization.
At first I confused this for disabling `GlobalNotices` - which A4A does use. But the two are different. 🚀

File: client/layout/index.jsx [hunk 1]
```diff
@@ -545,6 +549,7 @@ export default withCurrentRoute(
 			isGlobalSidebarCollapsed: shouldShowCollapsedGlobalSidebar && ! sidebarIsHidden,
 			isUnifiedSiteSidebarVisible: shouldShowUnifiedSiteSidebar && ! sidebarIsHidden,
 			isNewUser: isUserNewerThan( WEEK_IN_MILLISECONDS )( state ),
+			disableGlobalNotifications,
```
Comment (tyxla) [inline client/layout/index.jsx]: Is there a good reason for this to be a prop? Looks like all it does is a check against an environment check, and I don't think there's a way for the value to change without a full app refresh.

File: client/layout/index.jsx [hunk 2]
```diff
@@ -427,7 +429,7 @@ class Layout extends Component {
 				{ config.isEnabled( 'legal-updates-banner' ) && (
 					<AsyncLoad require="calypso/blocks/legal-updates-banner" placeholder={ null } />
 				) }
-				<GlobalNotifications />
+				{ ! shouldDisableGlobalNotifications && <GlobalNotifications /> }
```
Comment (tyxla) [inline client/layout/index.jsx]: I think this PR can be a one-liner if we do:

```suggestion
				{ ! isA8CForAgencies() && <GlobalNotifications /> }
```

See my comment below for info on why a prop is unnecessary.

**Review by tyxla — COMMENTED** [review]
I think we can simplify this a bit before shipping.

File: client/layout/index.jsx (hunk 2 above)
Comment (andrii-lysenko) [reply client/layout/index.jsx]: Thank you! Sounds good. My initial thought that we might use it somewhere else as well (for example Jetpack Manage). But I'm ok to move it as suggested.

**Pull request merged**